packages/bottom-tabs/src/views/BottomTabBar.tsx

```diff
@@ -53,51 +53,64 @@ export default function BottomTabBar({
   const { colors } = useTheme();
   const buildLink = useLinkBuilder();
 
-  const [dimensions, setDimensions] = React.useState(() => {
-    const { height = 0, width = 0 } = Dimensions.get('window');
-
-    return { height, width };
-  });
+  const focusedRoute = state.routes[state.index];
+  const focusedDescriptor = descriptors[focusedRoute.key];
+  const focusedOptions = focusedDescriptor.options;
 
-  const [layout, setLayout] = React.useState({
-    height: 0,
-    width: dimensions.width,
-  });
   const [keyboardShown, setKeyboardShown] = React.useState(false);
 
-  const [visible] = React.useState(() => new Animated.Value(1));
+  const tabBarHidden = focusedOptions.tabBarVisible === false;
+  const keyboardHidingTabBar = keyboardHidesTabBar && keyboardShown;
+  const shouldShowTabBar = !tabBarHidden && !keyboardHidingTabBar;
+  const prevShouldShowTabBarRef = React.useRef<boolean | undefined>();
 
-  const { routes } = state;
+  const [tabBarPartiallyHidden, setTabBarPartiallyHidden] = React.useState(
+    !shouldShowTabBar
+  );
+
+  const visibleRef = React.useRef<Animated.Value | undefined>();
+  if (!visibleRef.current) {
+    visibleRef.current = new Animated.Value(shouldShowTabBar ? 1 : 0);
```

Comment:
This was previously initialized via state. I should probably change it back, since I don't have a good understanding of how `Animated.Value` behaves in concurrent mode